ports/stm32/pyb_can.c

```diff
@@ -140,9 +140,57 @@ STATIC void pyb_can_print(const mp_print_t *print, mp_obj_t self_in, mp_print_ki
     }
 }
 
+STATIC void pyb_can_calc_bit_timing(mp_arg_val_t *args) {
+    enum { ARG_mode, ARG_extframe, ARG_prescaler, ARG_sjw, ARG_bs1, ARG_bs2, ARG_auto_restart, ARG_baudrate, ARG_sampling_point};
```

Comment:
These should definitely be global, so they are not duplicated.

Alternatively, could split this into a stand-alone function `uint32_t pyb_can_get_source_freq(void)`, and then put the rest of the searching logic in `pyb_can_init_helper` itself so the enum can stay private in that function.

Reply:
I like the split to `get_source_freq` and the search logic in `init_helper` more, will do that.